Source/BoundaryConditions/PML.cpp

```diff
@@ -759,9 +765,58 @@ PML::PML (const int lev, const BoxArray& grid_ba, const DistributionMapping& /*g
 }
 
 BoxArray
-PML::MakeBoxArray (const amrex::Geometry& geom, const amrex::BoxArray& grid_ba,
+PML::MakeBoxArray (const amrex::Box& regular_domain,
+                   const amrex::Geometry& geom, const amrex::BoxArray& grid_ba,
                    int ncell, int do_pml_in_domain,
                    const amrex::IntVect do_pml_Lo, const amrex::IntVect do_pml_Hi)
+{
+    if (regular_domain.numPts() == grid_ba.numPts()) {
+        return MakeBoxArray_single(regular_domain, grid_ba, ncell, do_pml_Lo, do_pml_Hi);
+    } else {
+        return MakeBoxArray_multiple(geom, grid_ba, ncell, do_pml_in_domain, do_pml_Lo, do_pml_Hi);
+    }
+}
+
+BoxArray
+PML::MakeBoxArray_single (const amrex::Box& regular_domain, const amrex::BoxArray& grid_ba,
+                          int ncell, const amrex::IntVect do_pml_Lo, const amrex::IntVect do_pml_Hi)
+{
+    BoxList bl;
+    for (int i = 0, N = grid_ba.size(); i < N; ++i) {
+        Box const& b = grid_ba[i];
+        for (OrientationIter oit; oit.isValid(); ++oit) { // Iteration over all faces
+            Orientation ori = oit();
+            const int idim = ori.coordDir();
+            bool pml_bndry = false;
+            if (ori.isLow() && do_pml_Lo[idim]) {
```

Comment:
Here and for line 793 too, would it be possible to add some short comment to clarify what these `if` conditions are checking exactly? Just to make the code a little bit easier to read for everyone. In particular, what do `ori.isLow()` and `ori.isHigh()` mean exactly, when true?

Reply:
```suggestion
            if (ori.isLow() && do_pml_Lo[idim]) {  // This is one of the lower side faces.
```

Reply:
Does the new comments/suggestions I added work?